修改模型类中使用addAll方法时，第三个replace参数无效问题

5ini99 opened the pull request:
文件
`thinkphp/library/think/db/Driver.php`
第950行
`$sql = 'INSERT INTO ' . $this->parseTable($options['table']) . ' (' . implode(',', $fields) . ') ' . implode(' UNION ALL ', $values);`
修改为
`$sql = (true === $replace ? 'REPLACE' : 'INSERT') . ' INTO ' . $this->parseTable($options['table']) . ' (' . implode(',', $fields) . ') ' . implode(' UNION ALL ', $values);`

kinosang commented:
呃，pr不是这么用的。

不会在本地修改的话，可以参考 https://github.com/top-think/think/blob/master/CONTRIBUTING.md 的`快速修改`小节

liu21st committed:
改进Validate类

liu21st committed:
简化validate类

liu21st committed:
修正

liu21st committed:
Validate类serialize方法支持二级

Commit:
Update Redis.php

修改缓存中的redis驱动不支持数组或对象数据存储的bug

kinosang committed:
Merge pull request #359 from 5ini99/patch-10

支持数组和对象

Commit:
瀹屽杽File缂撳瓨椹卞姩鍗曞厓娴嬭瘯

Commit:
澧炲姞Memcache缂撳瓨椹卞姩

Commit:
澧炲姞test缂撳瓨椹卞姩娴嬭瘯

liu21st committed:
Validate类支持读取模型类的属性定义完成验证

liu21st committed:
Merge branch 'master' of https://github.com/top-think/think

liu21st committed:
Merge pull request #358 from chunice/master

完善File缓存驱动单元测试

liu21st committed:
增加db_attr_case配置参数 用于配置 数据表字段大小写规则

liu21st committed:
Merge branch 'master' of https://github.com/top-think/think

liu21st committed:
自动填充移出Validate类

liu21st committed:
修正W助手函数

pakey committed:
taglib 增加一个U函数的tag

pakey committed:
代码格式化

liu21st committed:
Merge pull request #364 from pakey/master

在taglib中增加U函数的简写

liu21st committed:
修改File缓存驱动 默认path参数为 CACHE_PATH

liu21st committed:
修正memcached缓存驱动的有效期

liu21st committed:
sae模式配置调整

liu21st committed:
修正traits\model\Auto

liu21st committed:
Merge branch 'master' of https://github.com/top-think/think

liu21st committed:
版本更新

liu21st committed:
单元测试修正

liu21st committed:
readme更新

leunggamciu committed:
修复undefined index:extra

Commit:
优代部分代码；block标签内可以使用{__block__}来引用所继承模板中相应block标签的内容；
模板包含文件记录及更新时间直接写入缓存的模板中；改进标签别名的处理方式，别名不再定义方法；
去除include标签传参自动生成变量的代码；cx类增加function方法，用来生成匿名函数，结合{~$函数名()}可用于递归的实现。

liu21st committed:
Merge pull request #373 from oldrind/template

优代部分代码；block标签内可以使用{__block__}来引用所继承模板中相应block标签的内容；

liu21st committed:
Merge pull request #371 from leunggamciu/fix_undefined_index_extra

修复undefined index:extra

liu21st committed:
修正Response的一处问题

Commit:
修正模板缓存问题

liu21st committed:
Merge pull request #374 from oldrind/template

修正模板缓存问题

liu21st committed:
改进Trace日志驱动 支持配置 trace_tabs参数

liu21st committed:
改进session函数支持判断是否赋值操作 修正response类success方法

Commit:
修复模板更新后不更新缓存的问题

liu21st committed:
Merge pull request #375 from oldrind/template

修复模板更新后不更新缓存的问题

hugtale committed:
多维数组的点语法支持数字键值

liu21st committed:
Merge pull request #376 from hugtale/master

多维数组的点语法支持数字键值

liu21st committed:
修正 模板解析路径

liu21st committed:
修正Oracle驱动

liu21st committed:
改进View类

liu21st committed:
改进View类的模板引擎配置

hugtale committed:
替换T函数

liu21st committed:
View类engine参数更改为view_engine 避免冲突

liu21st committed:
修正单元测试

liu21st committed:
修正

liu21st committed:
Merge pull request #379 from hugtale/master

替换核心部分使用的T函数

liu21st committed:
View类优化

liu21st committed:
Merge branch 'master' of https://github.com/top-think/think

liu21st committed:
改进Config类

liu21st committed:
修正config类一处bug

liu21st committed:
改进View类对模板引擎的初始化

liu21st committed:
继续改进

liu21st committed:
改进部署模式下异常日志记录详细错误信息

liu21st committed:
改进异常页面显示

liu21st committed:
异常模板调整

Commit:
修改测试文件

Commit:
Revert "修改测试文件"

This reverts commit 58d37604892442ddd04dc012fe6aa5ebcb2e152c.

Commit:
瀹屽杽妗嗘灦鎯緥閰嶇疆鏂囦欢娉ㄩ噴

liu21st committed:
改进View类和Template类 简化模板主题（取消自动侦测） 当前主题作为view_path传入template 模板标签中可以使用 theme:controller/action 指定其他主题 或者 module@theme/controller/action 跨模块调用模板

liu21st committed:
模板引擎参数采用template二级配置

liu21st committed:
修正

liu21st committed:
Merge pull request #386 from chunice/master

完善框架惯例配置文件注释

liu21st committed:
改进View 避免手动初始化engine的时候 view_path为空值

liu21st committed:
Merge branch 'master' of https://github.com/top-think/think

liu21st committed:
修正View类模板定位

liu21st committed:
修正单元测试

liu21st committed:
改进模板变量赋值

leunggamciu committed:
修复错别字

liu21st committed:
Merge pull request #387 from leunggamciu/fix_typo

修复错别字

liu21st committed:
修正模板文件

liu21st committed:
修正

liu21st committed:
修正单元测试

liu21st committed:
修正单元测试

huangdijia committed:
优化2处写法

liu21st committed:
修正

liu21st committed:
Merge branch 'master' of https://github.com/top-think/think

liu21st committed:
修正template类的模板解析方法

Commit:
Merge branch 'master' of https://github.com/top-think/think

liu21st committed:
改进默认模板引擎的设置 如果没有设置 template.type 或者为空 默认为think

liu21st committed:
保证$_REQUEST正常取值

Commit:
Merge branch 'master' of https://github.com/top-think/think

Commit:
修正block标签的bug

kinosang committed:
Merge pull request #396 from oldrind/template

fix #390

hugtale committed:
修复内置模版引擎无法自动获取静态缓存的问题

liu21st committed:
Merge pull request #400 from hugtale/master

修复内置模版引擎无法自动获取静态缓存的问题

liu21st committed:
修正error_page配置无效的问题

liu21st committed:
Merge branch 'master' of https://github.com/top-think/think

Commit:
修复php标签合并时输出标签间的空格也被合并的bug

liu21st committed:
Merge pull request #405 from oldrind/template

修复php标签合并时输出标签间的空格也被合并的bug

leunggamciu committed:
修正phpdoc注释

kinosang committed:
Merge pull request #406 from leunggamciu/fix-phpdoc

修正phpdoc注释

liu21st committed:
改写Validate类和单元测试

liu21st committed:
Merge branch 'master' of https://github.com/top-think/think

liu21st committed:
修正Validate类message方法

liu21st committed:
改进model类的validate方法

liu21st committed:
validate类完善 controller和model类的validate方法支持传入提示信息 model类的validate方法支持指定验证器名

liu21st committed:
验证器类支持公共目录定义 修正Route类单一模块下面GET变量获取错误的BUG

liu21st committed:
验证类型规则支持使用别名 例如
'age'=>'>:10' 和 'age'=>'gt:10' 等效

liu21st committed:
改进model的validate方法

liu21st committed:
validate类的extend方法支持批量设置 修正单元测试

pwf0112 committed:
修正

kinosang committed:
Merge pull request #408 from pwf0112/master

修正 \think\Validate 初始化过程

liu21st committed:
验证方法和规则统一使用 驼峰法

yunwuxin committed:
添加console的MIT开源许可

liu21st committed:
改进validate类的unique规则

liu21st committed:
添加默认的规则提示信息

liu21st committed:
validate类增加setTypeMsg方法用于设置验证规则的默认提示信息

liu21st committed:
简化验证规则的默认提示信息定义

liu21st committed:
完善提示信息

liu21st committed:
修正unique验证规则

liu21st committed:
优化unique方法的查询

liu21st committed:
规则说明改进

liu21st committed:
删除 Model类的rule属性 验证规则支持直接传入提示信息 采用 [field,rule,msg] 的方式定义 所有验证规则只在 值不为空的时候进行验证 除非使用require规则

liu21st committed:
修正

liu21st committed:
修正单元测试

liu21st committed:
增加 default_validate 参数 用于设置默认的验证器名（用于controller和model的validate方法） 控制器类的validate方法增加callback参数在验证之前执行某个方法

liu21st committed:
controller和model类的validate方法支持指定场景 例如 $this->validate('User.edit');

liu21st committed:
validate的场景可以使用回调方法（支持闭包）而不是固定的字段

liu21st committed:
修正

liu21st committed:
改进validate类的验证规则错误提示信息获取

Commit:
修正Model类order方法一处bug，优化改进视图模型

liu21st committed:
validate自定义错误信息支持数组

liu21st committed:
改进validate类的unique验证规则的条件

liu21st committed:
改进unique验证规则的复合主键导致的错误 复合主键的情况直接使用条件判断表达式

liu21st committed:
修正一处错误

liu21st committed:
Merge pull request #413 from oldrind/master

修正Model类order方法一处bug，优化改进视图模型

liu21st committed:
提高runtime的计算精度 避免吞吐率出现Division by zero错误

liu21st committed:
改进PUT请求获取不到GET参数的问题

liu21st committed:
完善modelTest

liu21st committed:
修正单元测试

liu21st committed:
增加dbTest

liu21st committed:
修正debugTest

liu21st committed:
controllerTest增加validate测试

liu21st committed:
完善validateTest

liu21st committed:
完善validateTest

liu21st committed:
修正validate类的unique规则使用多字段检测

liu21st committed:
完善loaderTest

liu21st committed:
完善loaderTest

liu21st committed:
修正单元测试

liu21st committed:
简化validate

liu21st committed:
增加应用的环境变量配置文件 增加ROOT_PATH常量定义 runtime目录默认移到ROOT_PATH录下面

yunwuxin committed:
完善console

yunwuxin committed:
Merge remote-tracking branch 'origin/master'

liu21st committed:
环境变量读取改成getenv方法

liu21st committed:
默认的环境变量前缀 改成 PHP_

liu21st committed:
自动根据环境变量识别是否调试模式

liu21st committed:
调整Loader::addNamespace方法的位置 修正configTest

yunwuxin committed:
Merge remote-tracking branch 'origin/master'

yunwuxin committed:
修正console

kinosang committed:
框架核心独立

kinosang committed:
调整composer及入口文件

kinosang committed:
自动选择加载composer autoloader或tp loader

liu21st committed:
删除validate类make方法的第三个参数

liu21st committed:
Merge branch 'master' of https://github.com/top-think/think

liu21st committed:
Merge pull request #427 from top-think/split

将框架核心移动到独立仓库

liu21st committed:
清理文件和readme修改

kinosang committed:
更新 README.me

liu21st committed:
入口文件更改

liu21st committed:
Merge branch 'master' of https://github.com/top-think/think

yunwuxin committed:
修正composer安装，完善console模式

Commit:
增加 composer 安装方法说明

liu21st committed:
Merge pull request #436 from chunice/patch-1

增加 composer 安装方法说明

liu21st committed:
router文件移动位置

liu21st committed:
默认的入口文件 开启调试模式

liu21st committed:
readme修改

liu21st committed:
改进router.php

liu21st committed:
Merge branch 'master' of https://github.com/top-think/think

huangdijia committed:
移动router.php位置

yunwuxin committed:
preferred-install:dist

yunwuxin committed:
去除rewite的Multiviews选项

yunwuxin committed:
改进console模式

liu21st committed:
示例配置文件调整

liu21st committed:
取消APP_DEBUG 常量

liu21st committed:
调整配置文件

liu21st committed:
配置调整

liu21st committed:
修正配置

yunwuxin committed:
改进console模式 命令文件改为think

liu21st committed:
增加runtime目录

liu21st committed:
readme代码规范更新

yunwuxin committed:
修正样式

yunwuxin committed:
增加配置

liu21st committed:
配置文件调整

liu21st committed:
readme更新

lilwil committed:
更新配置

liu21st committed:
Merge pull request #476 from lilwil/master

更新配置

liu21st committed:
配置文件调整

liu21st committed:
配置调整

liu21st committed:
配置修改

liu21st committed:
配置文件调整

liu21st committed:
Merge branch 'master' of https://github.com/top-think/think

liu21st committed:
添加应用公共文件common.php

liu21st committed:
配置文件调整

liu21st committed:
build文件修改

yunwuxin committed:
增加单元测试示例

yunwuxin committed:
示例也放入think-testing包内

liu21st committed:
配置修改

liu21st committed:
更新配置文件

liu21st committed:
配置文件更新

liu21st committed:
添加应用默认行为扩展定义文件

lilwil committed:
一处注释

liu21st committed:
Merge pull request #519 from lilwil/master

一处注释

liu21st committed:
注释调整

kinosang committed:
指引开发者转向核心框架包

kinosang committed:
Update metadata badges

kinosang committed:
Merge pull request #525 from top-think/update-metadata-badges

Update metadata badges

kinosang committed:
Merge pull request #524 from top-think/rm-contributing-md

指引开发者转向核心框架包

kinosang committed:
Remove GitHub download metadata badges

kinosang committed:
Use stable version of ThinkPHP 5.0

liu21st committed:
readme文件修改

kinosang committed:
Replace shields badges with pugx badges

liu21st committed:
改进命令行入口文件

liu21st committed:
配置更新

xiasf committed:
添加：忽略框架目录

既然框架与应用仓库独立开来，那么应用仓库还是忽略框架目录比较好

liu21st committed:
Merge pull request #541 from xiasf/patch-1

添加：忽略框架目录

liu21st committed:
配置调整

liu21st committed:
配置更新

kinosang committed:
CI 自动打包

kinosang committed:
调整 CI 包安装

liu21st committed:
Apply fixes from StyleCI

liu21st committed:
Merge pull request #594 from top-think/analysis-qyRDeN

Apply fixes from StyleCI

liu21st committed:
配置调整

liu21st committed:
Merge branch 'master' of https://github.com/top-think/think

liu21st committed:
去除一些数据库配置 避免使用数据库驱动的时候影响

liu21st committed:
配置调整

liu21st committed:
更新

liu21st committed:
配置调整

tic8 committed:
框架引导文件改为绝对路径

liu21st committed:
去除无用配置

liu21st committed:
Merge branch 'master' of https://github.com/top-think/think

liu21st committed:
默认关闭调试模式

ganlvtech committed:
fixed #691

liu21st committed:
更新composer.json

liu21st committed:
改进composer.json

liu21st committed:
Merge branch 'master' of https://github.com/top-think/think

liu21st committed:
修正router

liu21st committed:
更新

liu21st committed:
Merge branch 'master' of https://github.com/top-think/think

liu21st committed:
composer调整

liu21st committed:
更新说明

liu21st committed:
更新

liu21st committed:
readme更新

liu21st committed:
配置调整

liu21st committed:
配置调整

liu21st committed:
调整

liu21st committed:
增加配置

liu21st committed:
添加更新日志

liu21st committed:
更新日志
